sdk/identity/azure-identity/src/test/java/com/azure/identity/ClientCertificateCredentialTest.java

```diff
@@ -193,29 +233,30 @@ public void testInvalidParameters() throws Exception {
         OffsetDateTime expiresOn = OffsetDateTime.now(ZoneOffset.UTC).plusHours(1);
 
         // mock
-        IdentityClient identityClient = PowerMockito.mock(IdentityClient.class);
-        when(identityClient.authenticateWithConfidentialClientCache(any())).thenReturn(Mono.empty());
-        when(identityClient.authenticateWithConfidentialClient(request)).thenReturn(TestUtils.getMockAccessToken(token1, expiresOn));
-        PowerMockito.whenNew(IdentityClient.class).withArguments(eq(TENANT_ID), eq(CLIENT_ID), isNull(), eq(pemPath), isNull(), isNull(), isNull(), isNull(), isNull(), eq(false), isNull(), any()).thenReturn(identityClient);
-
-        // test
-        try {
-            new ClientCertificateCredentialBuilder().clientId(CLIENT_ID).pemCertificate(pemPath).build();
-            fail();
-        } catch (IllegalArgumentException e) {
-            Assert.assertTrue(e.getMessage().contains("tenantId"));
-        }
-        try {
-            new ClientCertificateCredentialBuilder().tenantId(TENANT_ID).pemCertificate(pemPath).build();
-            fail();
-        } catch (IllegalArgumentException e) {
-            Assert.assertTrue(e.getMessage().contains("clientId"));
-        }
-        try {
-            new ClientCertificateCredentialBuilder().tenantId(TENANT_ID).clientId(CLIENT_ID).build();
-            fail();
-        } catch (IllegalArgumentException e) {
-            Assert.assertTrue(e.getMessage().contains("clientCertificate"));
+        try (MockedConstruction<IdentityClient> identityClientMock = mockConstruction(IdentityClient.class, (identityClient, context) -> {
+            when(identityClient.authenticateWithConfidentialClientCache(any())).thenReturn(Mono.empty());
+            when(identityClient.authenticateWithConfidentialClient(request)).thenReturn(TestUtils.getMockAccessToken(token1, expiresOn));
+        })) {
+            // test
+            try {
+                new ClientCertificateCredentialBuilder().clientId(CLIENT_ID).pemCertificate(pemPath).build();
+                fail();
+            } catch (IllegalArgumentException e) {
+                Assert.assertTrue(e.getMessage().contains("tenantId"));
+            }
```

Comment:
nit: This can be replaced with `assertThrows`

```java
IllegalArgumentException e = Assertions.assertThrows(IllegalArgumentException.class,
    () -> new ClientCertificateCredentialBuilder().clientId(CLIENT_ID).pemCertificate(pemPath).build());
Assert.assertTrue(e.getMessage().contains("tenantId"));
```

Reply:
I ended up backing off the junit 5 upgrade for now (I hit some other issues and it didn't fix the thing I hoped it would), but this is a good one to come back to.